apis/externalsecrets/v1beta1/secretstore_vault_types.go

```diff
@@ -124,7 +124,13 @@ type VaultAppRole struct {
 
 	// RoleID configured in the App Role authentication backend when setting
 	// up the authentication backend in Vault.
-	RoleID string `json:"roleId"`
+	RoleID string `json:"roleId,omitempty"`
```

Comment:
In order to make this optional to kubebuilder itself, you need to add a comment to flag this arg as optional:
```suggestion
        //+optional
	RoleID string `json:"roleId,omitempty"`
```